src/screens/SendConfirmationSheet.tsx

```diff
@@ -242,33 +262,149 @@ export default function SendConfirmationSheet() {
     return get(contacts, `${[toLower(toAddress)]}`);
   }, [contacts, toAddress]);
 
-  const [checkboxes, setCheckboxes] = useState([
-    {
-      checked: false,
-      label: lang.t(
-        'wallet.transaction.checkboxes.im_not_sending_to_an_exchange'
-      ),
-    },
-    {
-      checked: false,
-      label: lang.t(
-        'wallet.transaction.checkboxes.has_a_wallet_that_supports',
-        {
-          networkName: capitalize(network),
+  const uniqueTokenType = getUniqueTokenType(asset);
+  const isENS = uniqueTokenType === 'ENS';
+
+  const [checkboxes, setCheckboxes] = useState<Checkbox[]>(
+    isENS
+      ? ([
+          !hasClearProfileInfo(ensProfile) && {
+            checked: false,
+            id: 'clear-records',
+            label: lang.t(
+              'wallet.transaction.checkboxes.clear_profile_information'
+            ),
+          },
+          !doesNamePointToRecipient(ensProfile, toAddress) && {
+            checked: false,
+            id: 'set-address',
+            label: lang.t(
+              'wallet.transaction.checkboxes.point_name_to_recipient'
+            ),
+          },
+          doesAccountControlName(ensProfile) && {
+            checked: false,
+            id: 'transfer-control',
+            label: lang.t('wallet.transaction.checkboxes.transfer_control'),
+          },
+        ].filter(Boolean) as Checkbox[])
+      : [
+          {
+            checked: false,
+            id: 'not-sending-to-exchange',
+            label: lang.t(
+              'wallet.transaction.checkboxes.im_not_sending_to_an_exchange'
+            ),
+          },
+          {
+            checked: false,
+            id: 'has-wallet-that-supports',
+            label: lang.t(
+              'wallet.transaction.checkboxes.has_a_wallet_that_supports',
+              {
+                networkName: capitalize(network),
+              }
+            ),
+          },
+        ]
```

Comment:
Is there any chance to get the default state outside of the hook call? IMO it could be more readable if it was a one liner with some kind of a function call.